prepared_cache.go

```diff
@@ -2,60 +2,49 @@ package gocql
 
 import (
 	"github.com/gocql/gocql/internal/lru"
-	"sync"
 )
 
 const defaultMaxPreparedStmts = 1000
 
 // preparedLRU is the prepared statement cache
 type preparedLRU struct {
-	mu  sync.Mutex
 	lru *lru.Cache
```

Comment:
Why not use embedding and remove the proxy functions?

Reply:
I guess it can be fine here but theoretically it makes the external API wider since the methods get promoted.

Reply:
I added a couple of tests to just hammer the cache as the one in the semaphore test.
A bug was actually found when run with `-race` so this was a very good idea! Thanks @mmatczuk !